lib/ProductOpener/Export.pm

```diff
@@ -306,21 +333,42 @@ sub export_csv($) {
 
 						if (defined $tags_fields{$field}) {
 							if ((defined $product_ref->{$field . "_tags"}) and (scalar @{$product_ref->{$field . "_tags"}} > 0)) {
-								$populated_fields{$field} = sprintf("%08d", $group_number * 1000 + $item_number);
+								# Export the tags field in the main language of the product
+								$populated_fields{$group_prefix . $field} = sprintf("%08d", $group_number * 1000 + $item_number);
+								# Also possibly export the canonicalized tags
+								if ($export_canonicalized_tags_fields ) {
+									$populated_fields{$group_prefix . $field . "_tags"} = sprintf("%08d", $group_number * 1000 + $item_number) . "_tags";
+								}
 							}
 						}
 						elsif (defined $language_fields{$field}) {
 							if (defined $product_ref->{languages_codes}) {
 								foreach my $l (keys %{$product_ref->{languages_codes}}) {
 									if ((defined $product_ref->{$field . "_$l"}) and ($product_ref->{$field . "_$l"} ne "")) {
 										# Add language code to sort key
-										$populated_fields{$field . "_$l"} = sprintf("%08d", $group_number * 1000 + $item_number) . "_$l";
+										$populated_fields{$group_prefix . $field . "_$l"} = sprintf("%08d", $group_number * 1000 + $item_number) . "_$l";
 									}
 								}
 							}
 						}
-						elsif ((defined $product_ref->{$field}) and ($product_ref->{$field} ne "")) {
-							$populated_fields{$field} = sprintf("%08d", $group_number * 1000 + $item_number);
+						# Allow returning fields that are not at the root of the product structure
+						# e.g. ecoscore_data.agribalyse.score  -> $product_ref->{ecoscore_data}{agribalyse}{score}
+
+						# Special case for ecoscore_data.adjustments.origins_of_ingredients.value
+						# which is only present if the Eco-Score fields have been localized (done only once after)
+						elsif ($field eq "ecoscore_data.adjustments.origins_of_ingredients.value") {
+							if (deep_exists($product_ref, split(/\./, "ecoscore_data.adjustments.origins_of_ingredients.values"))) {
```

Comment:
you do this to add the "s" to `value` ? 
It's not explicit in your comment and hard to catch at first sight !